cmd/archeio/app/handlers.go

```diff
@@ -41,11 +41,17 @@ func MakeHandler(upstreamRegistry string) http.Handler {
 	blobs := newCachedBlobChecker()
 	doV2 := makeV2Handler(upstreamRegistry, blobs)
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// this registry only supports proxy image downloads / listing
+		// we do *not* want to deal with mutations, those are done against
+		// the backing registries by other tools
+		if r.Method != http.MethodGet && r.Method != http.MethodHead {
```

Comment:
unrelated tightening of the handler that I perhaps should have moved to another commit.

all of the endpoints we should support use one of these two methods https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints

POST/PUT/PATCH/DELETE are in the spec as well, but of course used for mutations (pushing images etc.), which we don't need or want to support here

Reply:
this needs test coverage, perhaps punt to another PR, it's not important.